pkg/ddc/thin/referencedataset/dataset.go

```diff
@@ -0,0 +1,30 @@
+package referencedataset
+
+import (
+	"fmt"
+	datav1alpha1 "github.com/fluid-cloudnative/fluid/api/v1alpha1"
+	"github.com/fluid-cloudnative/fluid/pkg/common"
+	"k8s.io/apimachinery/pkg/types"
+	"strings"
+)
+
+func getDatasetRefName(name, namespace string) string {
+	// split by '#', can not use '-' because namespace or name can contain '-'
+	return fmt.Sprintf("%s#%s", namespace, name)
```

Comment:
Since we already use `/` in `Dataset.spec` as the splitter between namespace and name, can we also use `/` for `status.DatasetRef`?